src/libmongoc/tests/unified/runner.c

```diff
@@ -236,6 +198,43 @@ test_runner_terminate_open_transactions (test_runner_t *test_runner)
    }
 }
 
+static test_runner_t *
+test_runner_new (void)
+{
+   test_runner_t *test_runner;
+   mongoc_apm_callbacks_t *callbacks;
+
+   test_runner = bson_malloc0 (sizeof (test_runner_t));
+   /* Create an client for internal test operations (e.g. checking server
+    * version) */
+   _mongoc_array_init (&test_runner->server_ids, sizeof (uint32_t));
+   callbacks = mongoc_apm_callbacks_new ();
+   mongoc_apm_set_topology_changed_cb (callbacks, on_topology_changed);
+   test_runner->internal_client = test_framework_client_new ();
+   mongoc_client_set_apm_callbacks (
+      test_runner->internal_client, callbacks, test_runner);
+   mongoc_client_set_error_api (test_runner->internal_client,
+                                MONGOC_ERROR_API_VERSION_2);
+   test_runner->topology_type =
+      get_topology_type (test_runner->internal_client);
+   server_semver (test_runner->internal_client, &test_runner->server_version);
+   test_diagnostics.test_runner = test_runner;
+
+   /* Terminate any possible open transactions. */
+   test_runner_terminate_open_transactions (test_runner);
```

Comment:
Note, `test_runner_new` was moved below the definition of `test_runner_terminate_open_transactions`. The only code change in this function was these two lines.